src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -89,9 +91,12 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
         _innerBuilder.Services.AddSingleton<ResourceLoggerService>();
 
         // Dashboard
+        _innerBuilder.Services.AddOptions<DashboardAuthenticationOptions>().Bind(Configuration.GetSection("Aspire:Dashboard:Authentication"));
```

Comment:
Can we enable the config binder source generator? That way we don't have to do this reflection at startup.

```xml
<EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>
```

in the .csproj.

Reply:
Ended up not using Bind at all. We are using custom environment variable names so I am just doing this via `PostConfigure<TOptions>(...)`.